keras_contrib/layers/convolutional.py

```diff
@@ -227,3 +228,234 @@ def get_config(self):
 Deconv3D = Deconvolution3D
 get_custom_objects().update({"Deconvolution3D": Deconvolution3D})
 get_custom_objects().update({"Deconv3D": Deconv3D})
+
+
+class CosineConvolution2D(Layer):
+    """Cosine Normalized Convolution operator for filtering windows of two-dimensional inputs.
+    Cosine Normalization: Using Cosine Similarity Instead of Dot Product in Neural Networks
```

Comment:
Better to move the references to the end of these comments, after the output shape, like https://github.com/fchollet/keras/blob/master/keras/layers/convolutional.py